sdk/azidentity/azidentity.go

```diff
@@ -34,18 +35,13 @@ func init() {
 	defaultTokenCredentialOpts = &TokenCredentialOptions{AuthorityHost: defaultAuthorityHostURL}
 }
 
-type internalAccessToken struct {
-	Token        string      `json:"access_token"`
-	RefreshToken string      `json:"refresh_token"`
-	ExpiresIn    json.Number `json:"expires_in"`
-	ExpiresOn    string      `json:"expires_on"`
-	NotBefore    string      `json:"not_before"`
-	Resource     string      `json:"resource"`
-	TokenType    string      `json:"token_type"`
+type tokenResponse struct {
+	token        *azcore.AccessToken
+	refreshToken string
 }
 
-// AuthenticationResponseError is a struct used to marshal responses when authentication has failed
-type AuthenticationResponseError struct {
+// AADAuthenticationFailedError is a struct used to marshal responses when authentication has failed
+type AADAuthenticationFailedError struct {
```

Comment:
Does this actually need to be exported?  If we don't expect consumers to `errors.As()` on this then it can be left internal.

Reply:
Leaving exported for now because AADAuthenticationFailedError exposes relevant fields for identity related errors